storage/objects/stream_file_upload.go

```diff
@@ -43,8 +43,21 @@ func streamFileUpload(w io.Writer, bucket, object string) error {
 	ctx, cancel := context.WithTimeout(ctx, time.Second*50)
 	defer cancel()
 
+	o := client.Bucket(bucket).Object(object)
+
+	// Add preconditions so that the upload may safely retry on transient errors.
+	// In this code sample, we assume the object does not yet exist.
+	o = o.If(storage.Conditions{DoesNotExist: true})
+
+	// To add preconditions for an existing object use the following instead:
```

Comment:
Replace "To add preconditions..." with "Sets a generation-match precondition for an existing object. The request to upload is successful if the object's generation number matches your precondition criteria."

Reply:
I think this doesn't make it clear that you would have to un-comment the line below to make this work, and that you have to choose one of the two options.

Reply:
I changed the wording. I'm not entirely happy with it so PTAL. I agree with Chris that we should make it clear that you have to choose one of the two options.

Reply:
@BrennaEpp Do you have a proposal for what wording you think would be best?

Reply:
@cbonnie I added above the following, which applies to both setting `DoesNotExist` and `GenerationMatch: ...` code: "Set a generation-match precondition. The request to upload is aborted if the object's generation number does not match your precondition criteria. This avoids race conditions and data corruption." 
Then for this line:
"If the live object already exists in your bucket, instead set a generation match condition using the live object's generation number as follows."